smoketest/graphqlupdaterotation_test.go

```diff
@@ -42,82 +42,100 @@ func TestGraphQLUpdateRotation(t *testing.T) {
 
 	var sched struct {
 		CreateSchedule struct {
-			ID        string
-			Rotations []struct{ ID string }
+			ID      string
+			Name    string
+			Targets []struct {
+				Target struct {
+					ID string
+				}
+			}
 		}
 	}
 	doQL(fmt.Sprintf(`
 		mutation {
-			createSchedule(input:{
-				name: "default",
-				description: "default testing",
-				time_zone: "America/Chicago",
-				default_rotation: {
-					type: daily,
-					start_time: "%s",
-    				shift_length:1,
-  				}
-			}){
+			createSchedule(
+				input: {
+					name: "default"
+					description: "default testing"
+					timeZone: "America/Chicago"
+					targets: {
+						newRotation: {
+							name: "foobar"
+							timeZone: "America/Chicago"
+							start: "2020-02-04T12:08:25-06:00"
+							type: daily
+						}
+						rules: {
+							start: "00:00"
+							end: "23:00"
+							weekdayFilter: [true, true, true, true, true]
+						}
+					}
+				}
+			) {
 				id
-				rotations {
-					id
+				name
+				targets {
+					target {
+						id
+					}
 				}
 			}
 		}
-	
-	`, time.Now().Format(time.RFC3339)), &sched)
+	`), &sched)
+
+	rotationID := sched.CreateSchedule.Targets[0].Target.ID
 
-	sID := sched.CreateSchedule.ID
-	var rotation struct {
-		CreateOrUpdateRotation struct {
-			Rotation struct {
-				ID   string
-				Name string
-			}
-		}
-	}
 	doQL(fmt.Sprintf(`
 		mutation {
-			createOrUpdateRotation(input:{
+			updateRotation(input:{
 				id: "%s",
-				name: "default",
-				start: "2017-08-15T19:00:00Z",
-				type: daily,
-				shift_length: 2,
-				schedule_id: "%s"
-			}){
-				rotation {
-					id
-					name
-				}
-			}
+				name: "new name",
+			})
```

Comment:
We should test updating all the fields, though previously we were only asserting on shift length.

Let's update it to check all the fields after updating, and can use the `assert` library to make it cleaner instead of the manual `if` checks.

```go
assert.Equal(t, "new name", .Name)
assert.Equal(t, 2, .ShiftLength)
```